---
title: "Haplotypes in the Chrom2 Spike Region"
author: "Hana Moyle"
subtitle: "Last Updated: 2024-07-16"
output: html_notebook
---

```{r libraries, echo = FALSE}
library(tidyverse)
library(ecaRbioinf) # run `remotes::install_github("eriqande/ecaRbioinf")` if you don't already have it
library(vcfR)
```

_References_
Haplo-rater Plots(006): https://eriqande.github.io/thompson-et-al-2020-chinook-salmon-migration-timing/006-haplo-raster-plots.html
Annotating variants (005):
https://eriqande.github.io/thompson-et-al-2020-chinook-salmon-migration-timing/005-annotating-variants-near-grev1l.html
Further Processing(004): https://eriqande.github.io/thompson-et-al-2020-chinook-salmon-migration-timing/004-prepare-haplotypes.html#further_processing
BEAGLE paper: https://www.cell.com/AJHG/fulltext/S0002-9297(07)63882-8

```{r sample_names, echo = FALSE}
sample <-read_csv("data/herring_metadata.csv") %>%
  mutate(newname = paste0(WATER_NAME, 
                          "-", 
                          NMFS_DNA_ID)) %>%
  mutate(newname = str_replace_all(newname, 
                                   " +", 
                                   "_"))
new_names <- sample %>%
  select(NMFS_DNA_ID, newname)

write_tsv(new_names, "data/phased/intermediates/new_names_for_vcf.txt",
          col_names = FALSE)
```


```{sh, eval = FALSE}
### if don't already have it, download the miniconda package ###
conda install conda-forge::mamba ## install mamba for some nice environments
source .zshrc ## source your config so that you can use conda and mamba
mamba create -c conda-forge -c bioconda -n vcftools vcftools ## installs the latest version of vcftools and names it vcftools through conda-forge and bioconda
conda install conda-forge::libzlib ## some of the necessary packages might be missing since we are just working on miniconda, so if it says  "RuntimeError: Found incorrect download: libzlib. Aborting" 
## run mamba create vcftools again
mamba create -c conda-forge -c bioconda -n bcftools bcftools ## installs latest version of bcftools

## now if you want to use bcftools and vcftools environments, you can run `conda activate myenv` (myenv=bcftools or vcftools) or shut down your entire shell
```


```{sh, eval = FALSE}
## now I'm going to reheader one of the full vcf files from beagle_regions/.../chr2_peak/vcf/
conda activate bcftools # activate bcftools environment
bcftools reheader -s intermediates/new_names_for_vcf.txt namaglfl.vcf.gz > intermediates/for-depths.vcf.gz
## the filter the sites
conda acitvate vcftools # activate vcftools environment
vcftools --gzvcf intermediates/for-depths.vcf.gz --min-alleles 2 --max-alleles 2 --max-missing 0.4 --maf 0.05 --out intermediates/namaglfl-filtered --recode
## attach the ancestral alleles from American Shad
 echo '##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral allele">' > aa.hdr
gunzip resources/GCF_018492685.1.fa.gz
bgzip resources/GCF_018492685.1.fa # make sure the reference genome is zipped correctly
bcftools +fill-from-fasta intermediates/namaglfl-filtered.recode.vcf -- -c AA -f resources/GCF_018492685.1.fa.gz -h aa.hdr > intermediates/almosst-ready.vcf # I downloaded the full A. Shad genome from UCSC Genome Browser
```


Extract Read Depths Information for Genotypes

```{r}
rfi <- read.vcfR("data/phased/intermediates/almosst-ready.vcf")

rfi_dephts <- vcfR2tidy(rfi)$gt %>%
  select(Indiv, POS, gt_DP) %>%
  mutate(DP = ifelse(is.na(gt_DP), 0, gt_DP)) %>%
  select(-gt_DP)

dp_means <- rfi_dephts %>%
  group_by(Indiv) %>%
  summarise(mean_depth =mean(DP)) %>%
  arrange(mean_depth)
dp_means

ggplot(dp_means, 
       aes(x = mean_depth)) +
  geom_histogram(binwidth = 0.1)
```

```{sh, eval = FALSE}
## all of my samples have pretty good read depth with the lowest being ~.8, so I'm going to keep all of them (other than the other two that had already been removed in previous quality checking)
tabix -f namaglfl-phased.vcf.gz # now I'm using the phased vcf from beagle_regions/../phased/ 
bcftools +fill-from-fasta namaglfl-phased.vcf.gz -- -c AA -f resources/GCF_018492685.1.fa.gz -h aa.hdr > namaglfl-pased-with-anc.vcf # and again attaching the ancestral states
bcftools reheader -s intermediates/new_names_for_vcf.txt namaglfl-pased-with-anc.vcf > namaglfl-phased-with-anc.vcf # and renaming the samples + getting the correct file name
```


```{r phased_data}
haps <- ecaRbioinf::vcf_haplos2tidy("data/phased/namaglfl-phased-with-anc.vcf", Anc = "Shad") # read in the phased vcf

samples <- sample %>%
  select(newname, STATE_F, WATERSHED, WATER_NAME, SPECIES, grouping_v3) # select important field from the metadata

## if we keep the sites that don't have an ancestral allele from Shad, mark those as NA so the derived allelic sites will be NA too
haps$fix <- haps$fix %>%
  mutate(AA = ifelse(AA == "N", NA, AA))

## straighten out the haplotype names
big_haps <- haps$tidy %>%
  filter(Indiv != "Shad") %>%
  left_join(haps$avd, by = c("ChromKey", 
                             "POS", 
                             "Indiv", 
                             "haplo")) %>%
  mutate(haplo_name = paste0(Indiv, "-", haplo)) %>%
  left_join(rfi_dephts, by = c("POS", "Indiv")) %>%
  left_join(.,
            samples, 
            by = c("Indiv" = "newname"), 
            relationship = "many-to-many")

## recode a column of alleles as M (for mid-atlantic) and N (for north atlantic)
mida_ones <- big_haps %>%
  filter(grouping_v3 == "MIDA") %>%
  group_by(POS, allele) %>%
  summarise(freq = n()) %>%
  filter(rank(-freq, ties = "first") == 1) %>%
  rename(mida_allele = allele) %>%
  ungroup()

big_haps2 <- big_haps %>%
  left_join(mida_ones, by = "POS") %>%
  mutate(alle2 = ifelse(allele == mida_allele, "M", "N"))

## join REF and ALT onto it, so that we know which allele is which
big_haps2 <- haps$fix %>%
  select(POS, REF, ALT) %>%
  left_join(big_haps, 
            ., 
            by = "POS") %>%
  select(ChromKey, POS, REF, ALT, everything())
```

Now, I can plot. Kind of. 

```{sh, eval = FALSE}
## intersected the gff(download from NCBI) file with the variants to a vcf with only one sample
bcftools view -s AP028690 namaglfl.vcf.gz > intermediates/one-sample-vcf-chr2-spike-ish.vcf.gz # i picked one of the mid-atlantic anadromous alewife ones
mamba create -c conda-forge -c bioconda -n bedtools bedtools # didn't have bedtools installed, so I installed it
conda activate bedtools
bedtools intersect -a intermediates/one-sample-vcf-chr2-spike-ish.vcf.gz -b resources/GCF_018492685.1_fAloSap1.pri.ncbiRefSeq.gtf.gz -wb > intermediates/variant-annotations.tsv # intersect them
```

```{r}
snp_anno <- read_tsv("data/phased/intermediates/variant-annotations-v2.tsv", 
                     col_names = c(
                       "CHROM", 
                       "POS", 
                       "dot1", 
                       "REF", 
                       "ALT", 
                       "VSCORE", 
                       "dot2", 
                       "dot3", 
                       "format", 
                       "a_sample", 
                       "CHROM2", 
                       "method", 
                       "what", 
                       "start", 
                       "stop", 
                       "dot4", 
                       "strand", 
                       "phase", 
                       "biggie"
                     )) # read in the SNP annotation file
snp_anno %>%
  count(what)

## first going to classify sites as belonging to genes, pseudo-genes, or non-genes
## then classify things within genes as exons, CDS, or other
 
gene_stuff_func <- function(what) {
  if(any(what == "gene")) {
    ret <- "gene"
  } else if (any(what == "pseudogene")) {
    ret <- "pseudogene"
  } else {
    ret <- "non-gene"
  }
  ret
}

gene_stuff <- snp_anno %>%
  group_by(CHROM, POS) %>%
  summarise(ginc = gene_stuff_func(what))

exon_stuff_func <- function(what) {
  if (any(what == "CDS")) {
    ret <- "CDS"
  } else if (any(what == "exon")) {
    ret <- "exon"
  } else {
    ret <- "non-exon"
  }
  ret
}

exon_stuff <- snp_anno %>%
  semi_join(gene_stuff %>% filter(ginc == "gene"), 
            by = c("CHROM", "POS")) %>% 
  group_by(CHROM, POS) %>%
  summarise(einc = exon_stuff_func(what))
  
```


